File: solutions/python/pangram/5/pangram.py
from string import ascii_lowercase
from typing import Set

LOWER_CASE_ASCII_SET: Set[str] = set(ascii_lowercase)


def is_pangram(sentence: str) -> bool:
    """
    Check if the given sentence is a pangram.

    A pangram is a sentence containing every letter of the alphabet at least once.

    Parameters:
    sentence (str): The sentence to check.

    Returns:
    bool: True if the sentence is a pangram, False otherwise.
    """
    return LOWER_CASE_ASCII_SET - set(sentence.lower()) == set()

File: solutions/elixir/pangram/1/lib/pangram.ex
defmodule Pangram do
  @doc """
  Determines if a word or sentence is a pangram.
  A pangram is a sentence using every letter of the alphabet at least once.

  Returns a boolean.

    ## Examples

      iex> Pangram.pangram?("the quick brown fox jumps over the lazy dog")
      true

  """

  @spec pangram?(String.t()) :: boolean
  def pangram?(sentence) do
      lowercase_charlist = sentence |> String.downcase |> String.to_charlist
      (for character <- ?a..?z, do: character in lowercase_charlist) |> Enum.all?
  end
end
